core/shared/src/main/scala-2/cats/effect/cpsinternal/AsyncAwaitStateMachine.scala

```diff
@@ -61,20 +62,20 @@ abstract class AsyncAwaitStateMachine[F[_]](
       // as inspecting the Succeeded outcome using dispatcher is risky on algebraic sums,
       // such as OptionT, EitherT, ...
       var awaitedValue: Option[AnyRef] = None
-      F.uncancelable { poll =>
+      F.race(stopSignal, F.uncancelable { poll =>
```

Comment:
I thought the same, but this comment indicates that maybe not :  https://github.com/typelevel/cats-effect-cps/pull/192#issuecomment-1899169045

Reply:
Indeed, this is referencing the mentioned workaround. Looking at the commit before, there is a green checkmark, but actually test is skipped (it is timing out!). See: https://github.com/typelevel/cats-effect-cps/actions/runs/7575437745/job/20632044148#step:10:64

The test gets stuck somewhere when waiting for the `defer.get` - and my debugging journey led me to believe that the dispatcher is working on some task and is not shutting down.

This not always happening, which is why I did not realize it initially when testing locally.

Reply:
What is interesting though:
- Why does this workaround (https://github.com/typelevel/cats-effect-cps/commit/99e8dbf1171dd97b31bd0abc4809d61ab5bacfe4) in the test fix the issue as well?
- When replacing the `defer.get` with `sleep` of around 300 millis, it also reliably works in the test - so the is dispatcher is not really getting stuck, but maybe something can deadlock here?

Reply:
I am wondering, whether it makes sense to first merge the change from `F.async_` to a cancellable `F.async`. Then we can discuss the remaining issues in a separate issue.

Reply:
@cornerman was the original test passing consistently after you changed the async call ? Because if that was the case, sure, please open a more atomic PR

Reply:
@Baccata :+1: Yes, it was. I have now updated the PR to only contain one fix. Also leaving the existing test in place and instead adding a new passing test - though this one has a `TODO` that we should tackle another time.